src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -1554,31 +1554,38 @@ LogicalResult ONNXBatchNormalizationInferenceModeOp::inferShapes(
 // Reshape
 //===----------------------------------------------------------------------===//
 
-LogicalResult ONNXReshapeOp::inferShapes(
-    std::function<void(mlir::Region &)> doShapeInference) {
+static LogicalResult verify(ONNXReshapeOp op) {
   // Cannot infer shape if no shape tensor is specified.
-  if (!data().getType().isa<RankedTensorType>())
+  if (!op.data().getType().isa<RankedTensorType>())
     return success();
 
-  if (!shape().getType().isa<RankedTensorType>())
+  if (!op.shape().getType().isa<RankedTensorType>())
     return success();
 
-  auto shapeTensorTy = shape().getType().cast<RankedTensorType>();
-  auto elementType = data().getType().cast<ShapedType>().getElementType();
+  auto shapeTensorTy = op.shape().getType().cast<RankedTensorType>();
 
   // Only rank 1 shape tensors are supported.
   if (shapeTensorTy.getShape().size() != 1)
-    return emitError("Shape tensor must have rank one");
-  int64_t outputRank = shapeTensorTy.getShape()[0];
+    return op.emitError("Shape tensor must have rank one");
 
-  // Shape tensor must have constant shape.
+  int64_t outputRank = shapeTensorTy.getShape()[0];
   if (outputRank < 0)
-    return emitError("Shape tensor must have constant shape");
+    return op.emitError("Shape tensor must have constant shape");
 
-  ONNXReshapeOpAdaptor operandAdaptor(*this);
-  ONNXReshapeOpShapeHelper shapeHelper(this);
+  ONNXReshapeOpAdaptor operandAdaptor(*&op);
+  ONNXReshapeOpShapeHelper shapeHelper(&op);
   if (failed(shapeHelper.computeShape(operandAdaptor)))
-    return emitError("Failed to scan Reshape parameters successfully");
+    return op.emitError("Failed to scan Reshape parameters successfully");
+
+  return success();
+}
+
+LogicalResult ONNXReshapeOp::inferShapes(
+    std::function<void(mlir::Region &)> doShapeInference) {
+
```

Comment:
you still have to keep the original code below... which was removed. Check a prior PR to see the code that was changed if you are not sure.
 
```
if (!op.shape().getType().isa<RankedTensorType>())
     return success();
```

Reply:
Done